src/skinview2d.ts

```diff
@@ -0,0 +1,56 @@
+// import { loadSkinToCanvas } from "skinview-utils";
+
+class SkinvViewer {
+
+  private canvas: HTMLCanvasElement;
+  private ctx: CanvasRenderingContext2D|null;
+  private domElement: HTMLElement;
+  private imagePath: string;
+  private skinImage: HTMLImageElement;
+
+  constructor(domELement: HTMLElement, imagePath: string) {
+   
+    this.domElement = domELement;
+    this.canvas = document.createElement("canvas");
+    this.ctx = this.canvas.getContext("2d");
+    this.imagePath = imagePath;
+
+    this.canvas.width = 400;
+    this.canvas.height = 400;
+
+    // add canvas to dom
+    this.domElement.appendChild(this.canvas);
+
+    this.skinImage = new Image();
+    this.skinImage.src = imagePath;
+
+    // texture loading
+    this.skinImage.crossOrigin = "anonymous";
+    this.skinImage.onerror = (): void => console.error("Failed loading " + this.skinImage.src);
+    this.skinImage.onload = (): void => {
+      // Not 100% sure I understand the benifit of this, right now it will just render the image
+      // directly to the canvas, which in the 2d skin viewer case is not what we want?
+      // loadSkinToCanvas(this.canvas, this.skinImage);
```

Comment:
My idea is to split this project into two parts:
 * One part provides some low-level functions to draw the player, like `model.ts` in skinview3d, for example:
    ```typescript
    drawHeadFront(ctx: CanvasRenderingContext2D, skinTexture: TextureCanvas, x: number, y: number, w:number, h: number);
    ```
 * Another part is `Skin2DViewer`, which provides high-level interfaces, like `viewer.ts` in skinview3d.

Reply:
```typescript
this.skinImage.crossOrigin = "anonymous";
this.skinImage.onerror = (): void => console.error("Failed loading " + this.skinImage.src);
this.skinImage.onload = (): void => { ...
```
I think this piece of code is a bit redundant (same code is in skinview3d).

I'm trying to extract the common part to skinview-utils, but I'm not satisfied with my current solution (see https://github.com/bs-community/skinview-utils/issues/3). Maybe it's a bit over design?